src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -648,6 +648,21 @@ public CalendarService getCalendarService() {
         return this.calendarService;
     }
 
+
+    public CalendarServiceData getCalendarDataService() {
+        CalendarServiceData calendarServiceData;
+        if (this.hasService(CalendarServiceData.class)) {
+            calendarServiceData = this.getService(CalendarServiceData.class);
+        } else {
+            calendarServiceData = new CalendarServiceData();
+        }
+        return calendarServiceData;
+    }
+
+    public void clearCachedCalenderService() {
+        this.calendarService = null;
```

Comment:
I agree with you @leonardehrenfried, but I think this is fine for now - removing it might cause a performance penalty in the `TimetableSnapshotSource` where it is used. Eventually, in OTP 2.2?, we will refactor the TransitLayer and restructure the calendar data, so putting too much effort into this now is not worth it.